test/test_02_01.py

```diff
@@ -0,0 +1,155 @@
+import os
+from typing import List
+
+import pytest
+
+from hm_1.tasks.task_02_01 import (
+    count_non_ascii_chars,
+    count_punctuation_chars,
+    get_longest_diverse_words,
+    get_most_common_non_ascii_char,
+    get_rarest_char,
+)
+
+
+@pytest.mark.parametrize(
+    ("file_name", "expected_result"),
+    [
+        (
+            os.path.dirname(__file__) + "/example1.txt",
+            [
+                "vorgebahnte",
+                "Betrachtung",
+                "ausführen",
+                "verbirgt",
+                "vielmehr",
+                "bedenkli",
+                "Waldgang",
+                "Ausflug",
+                "hinter",
+                "gefaßt",
+            ],
+        ),
+        (
+            os.path.dirname(__file__) + "/example2.txt",
+            [
+                "Gesellschaft",
+                "Entwicklung",
+                "Frauenfrage",
+                "praktische",
+                "wenngleich",
+                "Welträtsel",
+                "Optimismus",
+                "überhaupt",
+                "Forschung",
+                "Probleme",
+            ],
+        ),
+        (
+            os.path.dirname(__file__) + "/example3.txt",
+            [
+                "Gesellschaft",
+                "klassenlose",
+                "Inzwischen",
+                "entwickelt",
+                "Außenpoli",
+                "geworden",
+                "Gebieten",
+                "Planeten",
+                "soziale",
+                "gelöst",
+            ],
+        ),
+        (
+            os.path.dirname(__file__) + "/example4.txt",
+            [
+                "fragenstellende",
+                "ununterbrochen",
+                "objektiven",
+                "Problemen",
+                "beitragen",
+                "Wißbegier",
+                "geändert",
+                "Wahrheit",
+                "erfahren",
+                "Beitrag",
+            ],
+        ),
+        (
+            os.path.dirname(__file__) + "/example5.txt",
+            ["Mächte", "selbst", "Leser", "sichs", "wird", "sich", "Der", "an"],
+        ),
+    ],
+)
+def test_get_longest_diverse_words_works_right(
```

Comment:
Just replace `works_right` with the actual action.  `right` is a very vague term, each person has their own definition of `right`.

```python
def test_get_longest_diverse_words_return_lingest_diverse_words
```

Sounds a little bit strange, but 100% precise.